File: typescript/resistor-color-duo/resistor-color-duo.ts
export class ResistorColor {
  private colors: string[]


  public colorValues: any = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }

  constructor(colors: string[]) {

    if(colors.length < 2){
      throw Error('At least two colors need to be present');
    }

    this.colors = colors
    
  }

  value = (): number => {
    return parseInt(`${this.colorValues[this.colors[0]]}${this.colorValues[this.colors[1]]}`);
  }

}

File: dart/resistor-color-duo/lib/resistor_color_duo.dart
class ResistorColorDuo {

final colors = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
  ];

  int value(List<String> colors){
    return (this.colors.indexOf(colors[0])) * 10 + (this.colors.indexOf(colors[1]));
  }
}
